### DISCUSSION TITLE
Track user sessions

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAWeb#714] [Ideas]
As a player, I'd like to see how much time I spent playing each game.

We already know when a user starts playing a new game - this information appears in the global feed. This could be used to start a new session, and each time the Rich Presence ping is sent, the session could be extended.

For existing data, sessions could be estimated from achievement unlock timestamps. See https://github.com/Jamiras/RATools/wiki/Game-Mastery-Stats for one possible way to do that.

For RetroArch, we might be able to capture the start time the same way as above - does RetroArch do whatever is necessary to post a "Started Playing" message to the global feed? Then we'd have to update the session on each achievement unlock since RetroArch doesn't send Rich Presence pings.

Then, on the game page and user profile history blocks, session summary information could be displayed: "20hr10m in four sessions" or something similar.

Recommend flagging each session with the emulator and game MD5s (i.e. RANes 0.13, abcdef...) [this would probably require API changes]. A null emulator would indicate an estimated session, either from historical data or recent RetroArch data.

_I apologize if this was already suggested somewhere, I couldn't find it_

**Comment by kdecks** [comment]
This should be optionally private, private by default.
